src/aws/flb_aws_util.c

```diff
@@ -309,6 +310,15 @@ struct flb_http_client *request_do(struct flb_aws_client *aws_client,
         goto error;
     }
 
+    // Set max response buffer size for the http client. 
+    // Try setting resp.max_size to 0 to allow buffer to be resized. Zero means unlimited capacity as per requirement. 
```

Comment:
I would also put a comment near the #define to say that 0 means unlimited.

Reply:
added.